packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraAndroidCameraxPlugin.java

```diff
@@ -79,9 +80,17 @@ public void onAttachedToActivity(@NonNull ActivityPluginBinding activityPluginBi
         pluginBinding.getApplicationContext(),
         pluginBinding.getTextureRegistry());
     updateContext(pluginBinding.getApplicationContext());
-    processCameraProviderHostApi.setLifecycleOwner(
-        (LifecycleOwner) activityPluginBinding.getActivity());
-    systemServicesHostApi.setActivity(activityPluginBinding.getActivity());
+
+    Activity activity = activityPluginBinding.getActivity();
+
+    if (activity instanceof LifecycleOwner) {
```

Comment:
Is the `Activity` in the engine not always a `LifecycleOwner`?

Reply:
Actually, now that I think about it. This probably isn't guaranteed in an Add-to-app scenario.

Reply:
Yeah I think that's why the cast wasn't safe.